utils/logging/log.go

```diff
@@ -369,23 +389,29 @@ func (fw *fileWriter) Rotate() error {
 	return nil
 }
 
-func (fw *fileWriter) create(fileIndex int) (*bufio.Writer, *os.File, error) {
-	filename := filepath.Join(fw.config.Directory, fmt.Sprintf("%d.log", fw.fileIndex))
-	file, err := os.Create(filename)
+// Creates a file if it does not exist or opens it in append mode if it does
+func (fw *fileWriter) create() (*bufio.Writer, *os.File, error) {
+	filename := filepath.Join(fw.config.Directory, fmt.Sprintf("%s.log", fw.config.LoggerName))
+	file, err := os.OpenFile(filename, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
 	if err != nil {
 		return nil, nil, err
 	}
 	writer := bufio.NewWriter(file)
 	return writer, file, nil
 }
 
-func (fw *fileWriter) Initialize(config Config) error {
+// Initialize implements the RotatingWriter interface
+func (fw *fileWriter) Initialize(config Config) (int, error) {
 	fw.config = config
-	writer, file, err := fw.create(fw.fileIndex)
+	writer, file, err := fw.create()
 	if err != nil {
-		return err
+		return 0, err
 	}
 	fw.writer = writer
 	fw.file = file
-	return nil
+	fileSize, err := file.Seek(0, io.SeekEnd)
```

Comment:
https://golang.org/pkg/os/#File.Seek
> "The behavior of Seek on a file opened with O_APPEND is not specified."
Indeed the file has been opened with `os.O_APPEND|...` on line 395...

I wonder if `file.Stat()` then `FileInfo.Size()` is safer?

Reply:
Indeed